src/handler.lua

```diff
@@ -89,9 +89,14 @@ function JwtClaimsHeadersHandler:access(conf)
   kong.ctx.shared.jwt_claims = {}
   kong.ctx.shared.jwt_token = token
 
+  local anonymous_consumer = ngx.req.get_headers()["x-anonymous-consumer"]
```

Comment:
This checks the `x-anonymous-consumer` header that gets set by the `jwt` plugin. If this is true then the JWT is invalid and we should exit out early. This relies on an implementation detail of the JWT plugin. Ideally, this plugin would be able to operate in isolation, but it is a big pain to get the public key and decode the JWT from within this plugin. Also, any downstream services already rely on this implementation detail. And given it's part of the public interface of the JWT plugin I think it's ok to leave this as such.